pdf/pdfcanvas.cpp

```diff
@@ -164,12 +164,14 @@ void PDFCanvas::setDocument(PDFDocument* doc)
     }
 }
 
-qreal PDFCanvas::pagePosition(int index) const
+qreal PDFCanvas::pagePosition(int index, qreal offset) const
 {
     if( d->pages.count() == 0 )
         return 0.f;
 
-    return d->pages.value( index ).rect.y();
+    const QRectF& rect = d->pages.value( index ).rect;
+    qreal minOffset = 0., maxOffset = 1.;
+    return rect.y() + rect.height() * qBound(minOffset, offset, maxOffset);
```

Comment:
Minor detail, and fine as is too, but to me offset sounds like a discrete value, pixels for example. Also as qml side calculates contentX, maybe it could be simpler and more consistent, if this method would be called pageRectangle() and qml would decide position inside the page?